### PR TITLE
redis_exporter

### PR DESCRIPTION
**Pull request by dustinke11**

### CODE REVIEW

File: alicloud_redis_exporter.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	"time"
+
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/cms"
+	"github.com/patrickmn/go-cache"
+	"github.com/prometheus/common/log"
+)
+
+var (
+	accessKeyID     = os.Getenv("ALICLOUD_ACCESS_KEY")
```
Comment (bittopaz) [inline alicloud_redis_exporter.go]: We will also need RAM role to get the data

File: collector.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+			nil,
+		),
+		intranetOut: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, redisCollectorSubsystem, "intranet_out"),
+			"Read network rate",
+			[]string{"instanceId"},
+			nil,
+		),
+		failedCount: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, redisCollectorSubsystem, "failed_count"),
```
Comment (bittopaz) [inline collector.go]: We need to improve the separation of the `BuildFQName`

File: main.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	kingpin.Version(version.Print("alicloud_redis_exporter"))
+	kingpin.HelpFlag.Short('h')
+	kingpin.Parse()
+
+	log.Infoln("Starting alicloud_redis_exporter", version.Info())
+
+	http.HandleFunc(*metricsPath, handler)
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte(`<html>
+			<head><title>Node Exporter</title></head>
```
Comment (bittopaz) [inline main.go]: need to change the title

**Review by bittopaz — CHANGES_REQUESTED** [review]

File: alicloud_redis_exporter.go (hunk 1 above)
Comment (dustinke11) [reply alicloud_redis_exporter.go]: Resolved.

File: collector.go (hunk 2 above)
Comment (dustinke11) [reply collector.go]: Resolved.

File: main.go (hunk 3 above)
Comment (dustinke11) [reply main.go]: Resolved.

**Pull request merged**